provider/core/src/resource.rs

```diff
@@ -372,62 +373,113 @@ impl fmt::Display for ResourceOptions {
 
 impl Writeable for ResourceOptions {
     fn write_to<W: core::fmt::Write + ?Sized>(&self, sink: &mut W) -> core::fmt::Result {
-        let mut initial = true;
-        if let Some(variant) = &self.variant {
-            variant.write_to(sink)?;
-            initial = false;
-        }
-        if let Some(langid) = &self.langid {
-            if !initial {
-                sink.write_char('/')?;
-            }
-            langid.write_to(sink)?;
+        self.langid.write_to(sink)?;
+        if !self.keywords.is_empty() {
+            sink.write_str("-u-")?;
+            self.keywords.write_to(sink)?;
         }
         Ok(())
     }
 
     fn write_len(&self) -> LengthHint {
-        let mut length_hint = LengthHint::exact(0);
-        let mut initial = true;
-        if let Some(variant) = &self.variant {
-            length_hint += variant.write_len();
-            initial = false;
-        }
-        if let Some(langid) = &self.langid {
-            if !initial {
-                length_hint += 1;
-            }
-            length_hint += langid.write_len();
-        }
-        length_hint
+        self.langid.write_len()
+            + if !self.keywords.is_empty() { 3 } else { 0 }
+            + self.keywords.write_len()
```

Comment:
I remember why I did it this way:

```
error[E0308]: `if` and `else` have incompatible types
   --> provider/core/src/resource.rs:389:17
    |
386 |               + if !self.keywords.is_empty() {
    |  _______________-
387 | |                 self.keywords.write_len() + 3
    | |                 ----------------------------- expected because of this
388 | |             } else {
389 | |                 0
    | |                 ^ expected struct `LengthHint`, found integer
390 | |             }
    | |_____________- `if` and `else` have incompatible types
```

I'll fix it by using `LengthHint::exact(0)`